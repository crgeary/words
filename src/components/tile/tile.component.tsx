import { FC, PropsWithChildren } from "react";
import classNames from "classnames";
import { CharState } from "../../types/char-state";

export type TileProps = PropsWithChildren<{
    state?: CharState;
}>;

export const Tile: FC<TileProps> = ({ children, state = CharState.Idle }) => {
    return (
        <span
            className={classNames("inline-block w-full h-full p-2 text-center", {
                "bg-green-600 text-white": CharState.Correct === state,
                "bg-yellow-600 text-white": CharState.Present === state,
                "bg-slate-600 text-white": CharState.Absent === state,
                "bg-slate-300 text-gray-700": CharState.Idle === state,
            })}
        >
            {children && <span className="uppercase">{children}</span>}
        </span>
    );
};
