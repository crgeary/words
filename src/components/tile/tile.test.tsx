import { render } from "@testing-library/react";
import { Tile } from "./tile.component";

test("it should match snapshot", () => {
    const { container } = render(<Tile>A</Tile>);
    expect(container.firstChild).toMatchSnapshot();
});
