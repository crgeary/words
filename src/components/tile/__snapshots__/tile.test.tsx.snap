// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should match snapshot 1`] = `
<span
  class="inline-flex items-center border justify-center rounded w-full h-full p-2 bg-white border-violet-200 text-slate-600"
>
  <span
    class="uppercase font-medium text-xl"
  >
    A
  </span>
</span>
`;
